crates/router/src/connector/adyen/transformers.rs

```diff
@@ -1651,7 +1652,11 @@ fn get_recurring_processing_model(
         (_, Some(true)) => Ok((
             Some(AdyenRecurringModel::UnscheduledCardOnFile),
             None,
-            Some(format!("{}_{}", item.merchant_id, item.get_customer_id()?)),
+            Some(format!(
+                "{}_{}",
+                item.merchant_id,
+                item.get_customer_id()?.get_string_repr()
```

Comment:
Why do we need string representation of customer id can't we change it to the domain type itself.